azure-pipelines.yml

```diff
@@ -38,19 +38,41 @@ jobs:
     jobs:
       py38:
         image: [linux, vs2017-win2016, macOs]
+      py38-lower:
```

Comment:
please move them, separate group, as I did in my previous PR.

Reply:
I am afraid I lost that due to forced push. If you can do it again it would be great. After this I will add a condition to disable zipapp tests with lower constraints.

Reply:
Good

Reply:
Good

Reply:
What I do not know is how to disable the 3 zipapp tests when using lower, in a way that makes sense as I do not want to rely on environment name.

Any ideas?

Reply:
I don't think we should disable the zipapp tests 🤷‍♂ we should instead make them pass 👍

Reply:
Zipapp is part of specific packaging, if making them run would require bumping requirements it would be a bad idea as it will prevent other users from using virtualenv, when they do not have any need for zipapp.

I am inclined to believe that a correct approach would be to make zipapp an extra and to install it only on non-lower envs and to update the tests to skip on that case. If it is an extra, we can manually bump min versions there w/o affecting a normal install.

I did not look into why zipapp fails, maybe is something simple but sadly these days I am already overloaded and I have no background of using zipapp. maybe is something simple and we can avoid the more complex approach I mention in previous paragraph.

Reply:
Before we talk about any further options we need to understand why zipapp fails, it shouldn't.